src/utils/getArticleContents.js

```diff
@@ -0,0 +1,49 @@
+"use strict";
+
+const fluid = require("infusion");
+const jsdom = require("jsdom");
+const { JSDOM } = jsdom;
+
+/*
+ * Given page content markup, this function prepares a list of links to all headings of a given level
+ * followed by the page content with ID's injected for each linked heading.
+ *
+ * This work was undertaken to address GitHub issue #59:
+ * https://github.com/fluid-project/handbook.floeproject.org/issues/59
+ *
+ * @param {String} pageContent - the page content markup
+ * @param {String} summary - the summary text to display at the top of the article contents list
+ * @param {String} [headingsSelector] - a CSS selector to indicate which heading elements to list
+ *
+ * @return {String} - the combined string containing the list markup and modified page markup
+ */
+module.exports = (pageContent, summary, headingsSelector) => {
+    const dom = new JSDOM(pageContent);
+    const headings = dom.window.document.querySelectorAll(headingsSelector || "h2");
+
+    var articleContentsList = "";
+
+    for (const heading of headings) {
+        const listItemCssClass = "article-contents__list-item";
+        const linkCssClass = "article-contents__link";
+
+        var headingID = fluid.allocateSimpleId(heading); // also assigns the ID to that heading
```

Comment:
Rather than using this, it would probably be better to use [github-slugger](https://www.npmjs.com/package/github-slugger) as is done in [infusion-docs](https://github.com/fluid-project/infusion-docs/blob/04e01734587f8d22baa341a4045740460f29c82d/.eleventy.js#L37). In that way you'll add anchors to all the headings but also they'll be human readable too.

Reply:
You shouldn't need `fluid.allocateSimpleId` anymore if the ids are being added by GitHub-slugger. Instead you can just get the id from the element. Something like `heading.id`.